### PR TITLE
Refactoring code, unit tests for action.go added

### PR DESCRIPTION
**Pull request by bingobook02**

### CODE REVIEW

File: action/action_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	"github.com/damianoneill/nc-hammer/result"
+	"github.com/damianoneill/nc-hammer/suite"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_Execute(t *testing.T) {
+	var buff bytes.Buffer
+	var want string
+	ts1, _ := suite.NewTestSuite("../suite/testdata/test-suite.yml")             // testsuite with netconf actions
```
Comment (damianoneill) [inline action/action_test.go]: Never use identifiers with xx1, xx2, name the identifier so the user understands what it is for e.g. ts-valid or ts-invalid

File: action/action_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+	ts2, _ := suite.NewTestSuite("../suite/testdata/testsuite-fail-actions.yml") // testsuite with no netconf or sleep actions
+	myTests := []*suite.TestSuite{ts1, ts2}
+	start := time.Now()
+	resultChannel := make(chan result.NetconfResult)
+	handleResultsFinished := make(chan bool)
+	go result.HandleResults(resultChannel, handleResultsFinished, ts1)
+	for _, myTs := range myTests {
```
Comment (damianoneill) [inline action/action_test.go]: instead of myTs use the name of the type lower cased

File: action/action_test.go [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+			for _, a := range b.Actions {
+				log.SetFlags(log.Flags() &^ (log.Ldate | log.Ltime))
+				log.SetOutput(&buff)
+				if myTs == ts1 {
+					Execute(start, 0, ts1, a, resultChannel)
+					assert.True(t, (a.Sleep != nil) || (a.Netconf != nil)) // checks for netconf or sleep actions
+				} else {
+					Execute(start, 0, ts2, a, resultChannel)
+					got := buff.String()
+					want += "\n ** Problem with your Testsuite, an action in a block section has incorrect YAML indentation for its body, ensure that anything after netconf or sleep is properly indented **\n\n"
```
Comment (damianoneill) [inline action/action_test.go]: using string literals like this is very brittle, if someone changes the text (for a valid reason) then the test will need updated.  An improvement would be to check for contains in string for e.g. Problem.

File: cmd/run_test.go [hunk 4]
```diff
@@ -50,5 +87,11 @@ func Test_runTestSuite(t *testing.T) {
 	}
 	want += "Testsuite completed in "
 	want = strings.Replace(want, "\n", "", -1)
+
+	// testing handleBlocks
+	assert.Equal(t, myTs.Blocks[0].Type, "init")
+	assert.Equal(t, myTs.Blocks[1].Type, "concurrent")
```
Comment (damianoneill) [inline cmd/run_test.go]: does testing all three cases add anything to the test?

**Review by damianoneill — COMMENTED** [review]
Ahmed general comment - all this code added 1.3% coverage, you should question whether this is a good investment of your time?

File: cmd/run_test.go (hunk 4 above)
Comment (bingobook02) [reply cmd/run_test.go]: was't sure whether to add the three cases as handleBlocks got covered from being called from runTestSuite, but thought the function needed to be specifically checked

**Pull request merged**